File: homework-5/code/ArrayMin.go
// Function in action at http://play.golang.org/p/KU4Jn1Qkcx

func ArrayMin(a []int, min int) int {	
	if (len(a) == 0) {
		return min
	} else {
		if (a[0] < min) {
			min = a[0]
		}
	}
	return ArrayMin(a[1:], min)
}

File: homework-5/code/arraymin.c
// Function in action at http://codepad.org/D4reNoZ5

int arrayMin(int *a, int size, int min, int index) {
	if (index == size - 1) {
		return min;
	} else {
		min = min < a[index] ? min : a[index];
		index++;
	}
	return arrayMin(a, size, min, index);
}